### PR TITLE
fix(useFavicon): writable return type

### LINKED ISSUE #2031: useFavicon().value is readonly (TypeScript)
### Describe the bug

The documentation of [useFavicon](https://vueuse.org/core/usefavicon/#usefavicon) has the following example:

```typescript
const icon = useFavicon()
icon.value = 'dark.png'
```

Although this code generally works, it's not quite correct as the result of useFavicon() is a ComputedRef<>, where the value is readonly:

image

You can download the reproduction repo and execute "npm i" + "npm run build" to see the following error:

image

### Reproduction

https://github.com/stenet/vue-use-favicon

### System Info

```Shell
System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 Intel(R) Xeon(R) W-2133 CPU @ 3.60GHz
    Memory: 4.51 GB / 31.73 GB
  Binaries:
    Node: 17.4.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.18 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 7.20.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 103.0.5060.134
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.44)
    Internet Explorer: 11.0.19041.1566
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by chaii3**
closes #2031 

### Additional context

---

### What is the purpose of this pull request? 

- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [ ] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/core/useFavicon/index.ts [hunk 1]
```diff
@@ -9,15 +10,20 @@ export interface UseFaviconOptions extends ConfigurableDocument {
   rel?: string
 }
 
+export type UseFaviconValue = string | null | undefined
+export type UseFaviconGetter = () => UseFaviconValue
```
Comment (antfu) [inline packages/core/useFavicon/index.ts:14]: Prefer to use literals directly

File: packages/core/useFavicon/index.ts [hunk 2]
```diff
@@ -9,15 +10,20 @@ export interface UseFaviconOptions extends ConfigurableDocument {
...
+
 /**
  * Reactive favicon.
  *
  * @see https://vueuse.org/useFavicon
  * @param newIcon
  * @param options
  */
+export function useFavicon(newIcon: MaybeRef<UseFaviconValue>, options?: UseFaviconOptions): Ref<UseFaviconValue>
+export function useFavicon(newIcon: ComputedRef<UseFaviconValue> | UseFaviconGetter, options?: UseFaviconOptions): ComputedRef<UseFaviconValue>
```
Comment (antfu) [inline packages/core/useFavicon/index.ts:24]:
```suggestion
export function useFavicon(newIcon: MaybeComputedRef<UseFaviconValue>, options?: UseFaviconOptions): ComputedRef<UseFaviconValue>
```

Comment (chaii3) [reply packages/core/useFavicon/index.ts:23]: In this case the return type for Ref will be ComputedRef.
I mean for useFavicon(ref('str')): ComputedRef

**Review by antfu — APPROVED** [review]

**Comment by stef-work** [comment]
@antfu Sorry to get back to this again, but the function without parameters still returns a ComputedRef. Therefore, it still does not match the documentation.

image

**Pull request merged**